nueljoe/healthcare
# feature(orders): implement cart, orders and payment

#### What does this PR do?
Implement endpoints to manage carts, orders and payment for orders

#### Description of Task to be completed?
- implement the `GET /me/cart` endpoint to fetch user's cart.

- implement the `POST /me/cart/items` endpoint to add a product to a user's cart.

- implement the `DELETE /me/cart/items` endpoint to clear a user's cart.

- implement the `PATCH /me/cart/items/:itemId` endpoint to update a cart item.

- implement the `DELETE /me/cart/items/:itemId` endpoint to remove a single cart item.

- implement the `POST /me/cart/checkout` endpoint to checkout a user's cart, initiate a payment and create an order.

- implement the `GET /me/orders` endpoint to fetch a paginated list of orders ever created by the current user

- implement the `GET /orders` endpoint to fetch a paginated list of orders

- implement the `GET /orders/:reference`  endpoint to fetch the details of a single order

- implement the `PATCH /orders/:reference/delivery`  endpoint to manage the delivery status of an order

- implement the `PATCH /orders/:reference/cancel`  endpoint to cancel an order

#### How should this be manually tested?
- clone the repository
- cd into the project directory and run `cd api` to CD into the `api` directory
- see the README.md file in the `api` directory for the preliminary steps
- ensure that a mysql database in installed and running on your device
- see the postman documentation on `Cart` and `Orders`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
N/A

#### Screenshots (if appropriate)
N/A
#### Questions:
